src/modules/communication/internals/twilio-service.ts

```diff
@@ -5,29 +5,39 @@ import { SendSMSParams, ThirdPartyServiceError } from '../types';
 import SMSParams from './twilio-params';
 
 export default class TwilioService {
+  private static mock: boolean;
   private static twilio: Twilio;
 
-  public static initializeService(): void {
+  public static initializeService(mockMode: boolean): void {
+    this.mock = mockMode;
+
+    if (this.mock) return;
```

Comment:
@bbbneo333 

I would highly discourage this pattern.

This means we are adding code that is not desirable even in production build.

Recommend to mock appropriate third party calls using spyOn or any other framework as part of scenarios that get loaded before a E2E test is run. That also gives more flexibility for the caller to define the response from third party (success or error or anything else) depending on the test they want to write.